### ISSUE TITLE
Infinitivals

### ISSUE THREAD

**Issue by nschneid** [UniversalConceptualCognitiveAnnotation/docs#18]
Under "Infinitive 'to'" (p. 24) it would help to have analyses for

* __Inherent purpose infinitivals__, e.g. "a couch to sleep on" and "a procedure to ensure quality". My attempts:

   a_E couch_C [(IMP)_A to_F sleep_P [on_R (couch)_C]_A]_E

   [a_E procedure_P]_H to_L [(procedure)_A ensure_P quality_A]_H  
   *(This is an example of a sentence that causes [a cycle in AMR](https://github.com/amrisi/amr-guidelines/blob/master/amr.md#cycles).)*

* __For-to infinitivals__: The structure of these is controversial; some argue that *for* marks a subject of the infinitive clause, which could motivate:

   - **For the goalie to kick** a penalty shot in soccer is exciting: [For_R [the_E goalie_C]_C]_A to_F kick_P...
   - a procedure **for managers** to ensure quality: [a_E procedure_P]_H to_L [[for_R managers_C]_A (procedure)_A ensure_P quality_A]_H  

   Or it might be simpler to just treat *for* as another functional element:

   - For_F [the_E goalie_C]_A to_F kick_P...
   - [a_E procedure_P]_H for_L to_L [managers_A (procedure)_A ensure_P quality_A]_H

**Comment by nschneid** [comment]
Also, not an infinitival, but another way to express inherent purpose/function of something is with a gerund, like "the climbing wall". There is a sort of purpose-instrument interaction: climbing is the purpose of the wall, and in an act of climbing, the wall serves as an instrument. What is the best way to express this?

Contrast with "the flowering shrub", where it is a shrub that undergoes flowering—presumably:

the_E [flowering_P (shrub)_A]_E shrub_C

**Comment by omriabnd** [comment]
Re analyses above: I agree with your analysis of Inherent purpose infinitivals, about "for-to infinitvals" my hunch was always to analyze such cases as "for" being an R:
[ [For_R the_E goalie_C]_A to_F kick_P [the_E ball_C]_A ]_A is_F exciting_S 

Dotan, do you agree?

Re: the_E [flowering_P (shrub)_A]_E shrub_C
Yes, I think that would be the simplest solution.

** I didn't understand the example: [a_E procedure_P]_H for_L to_L [managers_A (procedure)_A ensure_P quality_A]_H

**Comment by nschneid** [comment]
> I didn't understand the example: [a_E procedure_P]_H for_L to_L [managers_A (procedure)_A ensure_P quality_A]_H

The idea is that there is a purposive relation between the procedure and ensuring quality, and the managers use the procedure (as an instrument) in the act of ensuring quality.

**Comment by dotdv** [comment]
> Re analyses above: I agree with your analysis of Inherent purpose infinitivals, about "for-to infinitvals" my hunch was always to analyze such cases as "for" being an R:
[ [For_R the_E goalie_C]_A to_F kick_P [the_E ball_C]_A ]_A is_F exciting_S

Dotan, do you agree?

Yes, I would also mark "for" as an R inside the A

**Comment by nschneid** [comment]
Decisions: a) The means within a purpose should be captured via linkage only, whereas an instrument is a remote participant in the goal scene. b) Adjunct infinitivals warrant an IMPLICIT unit where there is an unspecified agent. c) In a for-to infinitival, the "for" forms a participant with the subject. Thus 

a_E couch_C [(IMP)_A to_F sleep_P [on_R (couch)_C]_A]_E

a_E couch_C [[for_R somebody_C]_A to_F sleep_P [on_R (couch)_C]_A]_E

[a_E procedure_P]_H to_L [(IMP)_A ensure_P quality_A]_H

**Comment by jakpra** [comment]
I guess it's sort of implied from the alternatives @nschneid gave in the initial post, but just to clarify: In **for-to** constructions that have a **purposive** meaning, I assume we do

[they_A pay_P]_H to_L [[for_R you_C]_A ... use_P it_A]_H
"pay" doesn't seem like it should be remote in the "use" scene (at most as an adverbial, but I don't think we do that), "they" could potentially be a remote A as they are supporting the using.

[a_E procedure_P]_H to_L [(procedure)_A [for_R managers_C]_A ... ensure_P quality_A]_H

Right?

**Comment by nschneid** [comment]
With "pay" I would think what is paid for would be an A-scene:

I paid [for_R food_C]_A
I paid [for_R dinner_P]_A
I paid [to_F (I)_A eat_P dinner_A]_A
I paid [[for_R you_C]_A to_F eat_P dinner_A]_A

**Comment by jakpra** [comment]
Yes, I agree that this is borderline, but in STREUSLE we currently have "they pay for/INF you to/INF.P/Purpose", so maybe we could think about changing that to "to/INF"? Of course we don't have to smooth out every little difference between the two schemes, but I think this is a great opportunity for both approaches to inform each other.

Apart from this tricky case with "pay", another thing that arises from UCCA's new consistent treatment of for-to constructions is that we might want to change the "for"s to P (Agent when non-purposive and Agent~Beneficiary when purposive) in SNACS. I would like that much better than for/INF. I also don't think it breaks anything in the scheme, there even is one attested for/Agent~Beneficiary usage (for a completely different construction though).

**Comment by nschneid** [comment]
According to the guidelines, "pay for you TO..." should be `Theme~>Purpose`: http://flat.nert.georgetown.edu/en/infinitive_clauses/

The approach that feels most intuitive to me is to treat this special syntactic use of "for" as introducing a subject of the infinitival. It doesn't seem to contribute any lexical semantic content, as it doesn't alternate with other prepositions or license NPs that would not normally be subjects of the embedded verb (that I know of). But once we extend supersenses to subjects and objects, it would make sense to give the participant it marks a scene role.

**Comment by nschneid** [comment]
Relatedly: I should clarify the guidelines on this, but in SNACS we treat "need/take + resource + TO-infinitival" as `Purpose~>Purpose`. E.g. "I need some help TO survive the week" or "The water takes 10 minutes TO boil." However, "need" without a direct object is simply `` `i``: "I need TO tie my shoes."

As we discussed over email, the UCCA guidelines currently have "it took a lot of effort to_L win this fight", but the UCCA annotation of STREUSLE for a similar sentence treated "take" as a secondary verb (it seemed to take_D the hotel staff [quite a while]\_T to_F [quiet ... down]\_P them). I think it's a legitimate question whether an activity taking time/energy, or needing a resource to achieve something, is 1 scene or 2. Our SNACS annotation has not assumed a notion of secondary verbs, but maybe we should treat these modal verbs differently.

**Comment by jakpra** [comment]
Oh, I didn't look close enough, the example in the corpus is actually `Purpose~Theme`, which according to the guidelines is also wrong (flipped) then.

I don't have a super strong opinion on "for", but while I agree that it doesn't contribute a lot of lexical content (except for maybe marking the focus; there is a similar and probably related construction in German which has a clear focus meaning contributed by "für"), the fact that it doesn't alternate with other prepositions alone doesn't necessarily mean that it can't get any supersense or even be P: For example, the by-passive is our prototype of Agent, but it's a very grammatical construction that requires this specific preposition.

I think, if we go really in-depth here (and compare with the parallel constructions in German), we could probably differentiate cases where "for" is purely a complementizer that introduces the infinitival clause from cases where it rather marks the subject as either Focus, Beneficiary, or Agent. I don't know if we want that.

But for the alignment of SNACS and UCCA I think it would be worth trying to find a more consistent treatment. For example, if we insist that "for" is purely grammatical and doesn't carry any relational meaning, then it should be F in UCCA. If we say that it does mark the participant (not necessarily agent) of the infinitival clause, then it should get a supersense. 

The analysis of "to" in the "pay" examples looks good to me now: Our use of the Theme scene role should be compatible with F.
